saleor/discount/migrations/tasks/saleor3_16.py

```diff
@@ -278,28 +292,54 @@ def migrate_translations(sale_ids, saleid_promotion_map):
 
 
 def migrate_checkout_line_discounts(sale_ids, rule_by_channel_and_sale):
-    if checkout_line_discounts := CheckoutLineDiscount.objects.filter(
+    checkout_line_discounts = CheckoutLineDiscount.objects.filter(
         sale_id__in=sale_ids
-    ).select_related("line__checkout"):
-        for checkout_line_discount in checkout_line_discounts:
+    ).exclude(type="promotion")[:CHECKOUT_LINE_DISCOUNT_BATCH_SIZE]
+    discount_line_ids = list(checkout_line_discounts.values_list("id", flat=True))
+    if discount_line_ids:
+        line_discounts = CheckoutLineDiscount.objects.filter(
+            id__in=discount_line_ids
+        ).select_related("line__checkout")
```

Comment:
Just a comment, as this line is fine.

If there were issue with memory limit, you can always specify here what fields exactly from `checkout` are needed with `prefetch_related` or by using `only` with `select_related`.

Reply:
Done